components/tidb_query/src/codec/table.rs

```diff
@@ -73,6 +73,40 @@ pub fn check_table_ranges(ranges: &[KeyRange]) -> Result<()> {
     Ok(())
 }
 
+#[inline]
+pub fn check_record_key(key: &[u8]) -> Result<()> {
+    check_key_type(key, RECORD_PREFIX_SEP)
+}
+
+#[inline]
+pub fn check_index_key(key: &[u8]) -> Result<()> {
+    check_key_type(key, INDEX_PREFIX_SEP)
+}
+
+/// `check_key_type` checks if the key is the type we want, `wanted_type` should be
+/// `table::RECORD_PREFIX_SEP` or `table::INDEX_PREFIX_SEP` .
+#[inline]
+pub fn check_key_type(key: &[u8], wanted_type: &[u8]) -> Result<()> {
```

Comment:
`read_bytes` will return `Error::Io` if `key` is not long enough. Should we `.map_err` here? @breeswish

Reply:
It seems that  `codec::Error` has implemented `From<io::Error> for Error`, so it can be converted automatically.

Reply:
Yes. It can be converted automatically, otherwise you are not able to pass the compile. But I wonder if is better to return an Evaluate Error rather than io error.